### PR TITLE
Okta authentication sample

### PR DESCRIPTION
**Pull request by kevinthecheung**

### CODE REVIEW

File: okta-auth/public/index.html [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+                        </tr>
+                        <tr>
+                            <th>Email address</th>
+                            <td>${user.email}</td>
+                        </tr>
+                        <tr>
+                            <th>Unique ID</th>
+                            <td>${user.uid}</td>
+                        </tr>
+                    </table>`;
```
Comment (rachelmyers) [inline okta-auth/public/index.html:95]: Could update this to
``` 
`Hi, ${user.displayName}! Your email address is
 ${user.email} and your UID is ${user.uid}.`;
```

File: okta-auth/setup.js [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+
+You can download your project's service account key from the Firebase console:
+https://console.firebase.google.com/project/_/settings/serviceaccounts/adminsdk
+
+Note that this value is required only when using the emulator.
+> `;
+
+const EXAMPLE_ENDPOINT = (
+        DEPLOYING ? `
+When you deploy the Cloud Function. the endpoint looks like:
```
Comment (rachelmyers) [inline okta-auth/setup.js:46]: Should the period here be a comma? or just go away?

File: okta-auth/README.md [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+3.  Make sure the Firebase CLI tool is set to use your Firebase project:
+
+    ```
+    $ cd functions-samples/okta-auth
+    okta-auth$ firebase login
+    okta-auth$ firebase use <YOUR_FIREBASE_PROJECT_ID>
+    ```
+
+4.  Run `setup.js -d` from the Firebase project directory:
```
Comment (rachelmyers) [inline okta-auth/README.md:106]: Maybe explain that the `-d` flag is used in this script to make sure the Cloud Functions environment is set up before a deploy.

File: okta-auth/functions/index.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+ * Implements the `/api/firebaseCustomToken` endpoint, which is responsible for
+ * validating Okta access tokens and minting Firebase custom authentication
+ * tokens.
+ */
+const express = require('express');
+const app = express();
+
+// For local testing, use GOOGLE_APPLICATION_CREDENTIALS from `.env` instead of
+// the value set by the emulator.
+const envCfg = require('dotenv').config();
```
Comment (rachelmyers) [inline okta-auth/functions/index.js:11]: Maybe explain that the setup script will populate the `.env` file?

File: okta-auth/public/index.html (hunk 1 above)
Comment (kevinthecheung) [reply okta-auth/public/index.html:95]: I like the table when code snippet length isn't an issue

**Pull request merged**